### ISSUE TITLE
Print values for empty JSON fields in FileAnnotations

### ISSUE THREAD

**Issue by devkvlt** [bufbuild/buf#1930]
### Problem

When we run `buf lint --error-format=json` on an empty proto file, we get the following JSON output:

```json
{
  "path": "path/to/file.proto",
  "type": "PACKAGE_DEFINED",
  "message": "Files must have a package defined."
}
{
  "path": "path/to/file.proto",
  "type": "SYNTAX_SPECIFIED",
  "message": "Files must have a syntax explicitly specified. If no syntax is specified, the file defaults to \"proto2\"."
}
```

This output is missing 4 fields `start_line`, `start_column`, `end_line` and `end_column` which makes consuming this JSON by third party tools problematic as one may easily forget to check if these fields are not null before assigning them as numeric values. 

### Proposition

Add the missing fields with value 1 so that the output becomes:

```json
{
  "path": "path/to/file.proto",
  "start_line": 1,
  "start_column": 1,
  "end_line": 1,
  "end_column": 1,
  "type": "PACKAGE_DEFINED",
  "message": "Files must have a package defined."
}
{
  "path": "path/to/file.proto",
  "start_line": 1,
  "start_column": 1,
  "end_line": 1,
  "end_column": 1,
  "type": "SYNTAX_SPECIFIED",
  "message": "Files must have a syntax explicitly specified. If no syntax is specified, the file defaults to \"proto2\"."
}
```

**Comment by bufdev** [comment]
I can see the logic here, but there's also the inverse logic of this: there is no line 1 for an empty file, so you are indicating something that doesn't exist. You could make the argument that we already do this for the text format though, where we print `1:1`, so changing this would make this consistent. Perhaps omitting empty values would be better, but we'd want to make sure we do this consistently across all of our JSON outputs from `buf`. What specific third party tool are you referencing that makes this difficult?

@akshayjshah @timostamm interested if you have any input as well.

For reference, if we do do this: We likely just want to change `printFileAnnotationAsJSON` in `private/bufpkg/bufanalysis/print.go`.

**Comment by bufdev** [comment]
I'm wondering if we should though - the more I think about it, it's inconsistent with the text format (which prints 1:1).

I think I might be coming around to the argument presented in the description of this issue, but am worried this may be a breaking change, although it's in a gray area.

**Comment by timostamm** [comment]
For JSON, I can totally see myself looking at the output and writing code that ingests it, only to find out later about unknown locations. 

It looks like `PACKAGE_DEFINED` and `FILE_LOWER_SNAKE_CASE` are the only two rules where we don't provide a location. In my mind it's fair to use 1:1 for them as well, and have the same values for the text format and for JSON, which is the least surprising behavior.

**Comment by devkvlt** [comment]
@bufdev 
> there is no line 1 for an empty file, so you are indicating something that doesn't exist

That's true, but this is sort of like a convention that most other tools use as well. The thing is, even if line 1 doesn't exist, when you open a file the cursor is already on line 1 so I can't see this being an issue.

> What specific third party tool are you referencing that makes this difficult?

Just auto linters for code editors.

@akshayjshah 
> If the problem is null handling, would emitting explicit zeros help?

Yes, but I'm wondering if it shouldn't be 1 rather than 0.

**Comment by saquibmian** [comment]
FWIW, for format `text` we output `1:1`, but for format `msvs` we output `1:0`. At least in the `msvs` case, this should be either `<empty>`, `1` or `1:1` [as per the their docs](https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-diagnostic-format-for-tasks?view=vs-2022) (col 0 isn't valid). We also don't print the end line/column, even though we're allowed to.

For the JSON format, I think we should not emit it as `1`. In my view, the JSON format should be the lowest common denominator between all the formats, so it needs to be able to capture the case where there is no location. This way if there is a format that we don't support, users can pipe the output and generate it themselves. I think we should drop the `omitempty` and emit as `null`, but that's only because I dislike when tools output mixed-shaped JSON.

**Comment by bufdev** [comment]
We've merged this in #1970, we're going to do a micro release 1.16.1 today to get this out.
